engine/src/main/scala/cromwell/server/CromwellShutdown.scala

```diff
@@ -0,0 +1,222 @@
+package cromwell.server
+
+import java.util.concurrent.atomic.AtomicBoolean
+
+import akka.Done
+import akka.actor.{ActorRef, ActorSystem, CoordinatedShutdown, _}
+import akka.http.scaladsl.Http
+import akka.pattern.{AskTimeoutException, GracefulStopSupport}
+import akka.routing.Broadcast
+import akka.stream.ActorMaterializer
+import akka.util.Timeout
+import cats.instances.future._
+import cats.syntax.functor._
+import com.typesafe.config.ConfigFactory
+import cromwell.engine.workflow.WorkflowManagerActor.{AbortAllWorkflowsCommand, PreventNewWorkflowsFromStarting}
+import cromwell.services.SingletonServicesStore
+import cromwell.util.GracefulShutdownHelper.ShutdownCommand
+import org.slf4j.LoggerFactory
+
+import scala.concurrent.duration._
+import scala.concurrent.{ExecutionContext, Future}
+import scala.util.{Failure, Success}
+
+/**
+  * Collection of methods and objects used to control Cromwell graceful shutdown process.
+  */
+object CromwellShutdown extends GracefulStopSupport {
+  private val logger = LoggerFactory.getLogger("CromwellShutdown")
+  
+  // Includes DB writing actors, I/O Actor and DockerHashActor
```

Comment:
IMO these would read better as `FooPhase` vs `PhaseFoo`. It took me a moment to realize what this meant and I'm reading the PR.